OpenRA.Mods.Common/Traits/World/WeatherOverlay.cs

```diff
@@ -0,0 +1,327 @@
+﻿#region Copyright & License Information
+/*
+ * Copyright 2007-2015 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation. For more information,
+ * see COPYING.
+ */
+#endregion
+
+using System;
+using System.Collections.Generic;
+using System.Drawing;
+using System.Linq;
+using OpenRA.Graphics;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Common.Traits
+{
+	[Desc("Adds a particle-based overlay.")]
+	public class WeatherOverlayInfo : ITraitInfo
+	{
+		[Desc("Factor for particle density. A smaller value increases the number of particles spawned.")]
+		public readonly int ParticleDensityFactor = 1311;
+
+		[Desc("The wind will change over time by the values of WindStrength, or just stick to the first value.")]
+		public readonly bool ChangingWind = true;
+
+		[Desc("Works only if ChangingWind is enabled. The values of the wind intensity and direction. ")]
+		public readonly int[] WindStrength = { -5, -3, -2, 0, 2, 3, 5 };
```

Comment:
Following in from my earlier questions/comments, please use an intuitive unit for this (such as pixels per tick - float values are fine since these are visual) instead of arbitrary units with an arbitrary divisor in the code.